src/celeritas/em/distribution/BetheBlochEnergyDistribution.hh

```diff
@@ -0,0 +1,127 @@
+ * differential cross section can be factorized as
+ * \f[
+   \difd{\sigma}{T} = C f(T) g(T)
+ * \f]
+ * with \f$ T \in [T_{cut}, T_{max}] \f$, where \f$ f(T) = \frac{1}{T^2} \f$,
+ * \f$ g(T) = 1 - \beta^2 \frac{T}{T_max} + s \frac{T^2}{2 E^2} \f$, \f$ T \f$
+ * is the kinetic energy of the incident particle, \f$ E \f$ is the total
+ * energy, and \f$ s \f$ is 0 for spinless particles and 1 otherwise. The
+ * energy is sampled from \f$ f(T) \f$ and accepted with probability \f$ g(T)
+ * \f$.
+ */
+class BetheBlochEnergyDistribution
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using Energy = units::MevEnergy;
+    using Mass = units::MevMass;
+    //!@}
+
+  public:
+    // Construct with incident and exiting particle data
+    inline CELER_FUNCTION
+    BetheBlochEnergyDistribution(ParticleTrackView const& particle,
+                                 Energy electron_cutoff,
+                                 Mass electron_mass);
+
+    // Sample the exiting energy
+    template<class Engine>
+    inline CELER_FUNCTION Energy operator()(Engine& rng);
+
+    //! Minimum energy of the secondary electron [MeV].
+    CELER_FUNCTION Energy min_secondary_energy() const { return min_energy_; }
+
+    //! Maximum energy of the secondary electron [MeV].
+    CELER_FUNCTION Energy max_secondary_energy() const { return max_energy_; }
+
+  private:
+    // Incident partcle mass
+    real_type inc_mass_;
+    // Square of fractional speed of light for incident particle
+    real_type beta_sq_;
+    // Minimum energy of the secondary electron [MeV]
+    Energy min_energy_;
+    // Maximum energy of the secondary electron [MeV]
+    Energy max_energy_;
+};
+
+//---------------------------------------------------------------------------//
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with incident and exiting particle data.
+ */
+CELER_FUNCTION
+BetheBlochEnergyDistribution::BetheBlochEnergyDistribution(
+    ParticleTrackView const& particle,
+    Energy electron_cutoff,
+    Mass electron_mass)
+    : inc_mass_(value_as<Mass>(particle.mass()))
+    , beta_sq_(particle.beta_sq())
+    , min_energy_(electron_cutoff)
+    , max_energy_(detail::calc_max_secondary_energy(particle, electron_mass))
+{
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Sample secondary electron energy.
+ */
+template<class Engine>
+CELER_FUNCTION auto BetheBlochEnergyDistribution::operator()(Engine& rng)
+    -> Energy
+{
+    InverseSquareDistribution sample_energy(value_as<Energy>(min_energy_),
+                                            value_as<Energy>(max_energy_));
+    real_type energy;
+    do
+    {
+        // Sample 1/E^2 from Emin to Emax
+        energy = sample_energy(rng);
+        /*!
+         * \todo Adjust rejection functions if particle has positive spin
+         */
```

Comment:
So the MuBB model is this the same as the BetheBloch but with rejection functions? The MuBBEnergyDistribution could use a touch more documentation describing the difference between these... it might also be fun to rig up a little test that lets us plot the exiting energy distributions for a couple of different incident particle energies.

Reply:
Right, for muons it's just different rejection functions (for hadrons after sampling the energy the Bethe-Bloch model will also sample whether or not to produce a high energy delta ray). A distribution test is a good idea.